tests/test_olio_grid_functions.py

```diff
@@ -0,0 +1,236 @@
+
+    # --------- Assert----------
+    np.testing.assert_almost_equal(tri[0, :, :, 0][0][0], k_face_centre_points[0])
+    np.testing.assert_almost_equal(tri[0, :, :, 0][1][0], k_face_centre_points[1])
+    np.testing.assert_almost_equal(tri[1, :, :, 0][0][0], j_face_centre_points[0])
+    np.testing.assert_almost_equal(tri[1, :, :, 0][1][0], j_face_centre_points[1])
+    np.testing.assert_almost_equal(tri[2, :, :, 0][0][0], i_face_centre_points[0])
+    np.testing.assert_almost_equal(tri[2, :, :, 0][1][0], i_face_centre_points[1])
+
+
+def test_actual_pillar_shape(tmp_path):  # TODO: test with more complicated grid geometries
+
+    # --------- Arrange----------
+    epc = os.path.join(tmp_path, 'grid.epc')
+    model = rq.new_model(epc)
+
+    # create a basic block grid
+    dxyz = (10.0, 10.0, 10.0)
+    grid = grr.RegularGrid(model, extent_kji = (3, 3, 2), title = 'grid1', origin = (0.0, 0.0, 1000.0), dxyz = dxyz)
+
+    # --------- Act----------
+    pillar_shape = actual_pillar_shape(pillar_points = grid.corner_points())
+
+    # --------- Assert----------
+    assert pillar_shape == 'vertical'
+
+
+def test_determine_corp_ijk_handedness():
+
+    # --------- Arrange----------
+    corner_points_right = np.array([[[[[[[0., 0., 0.], [10., 0., 0.]], [[0., -10., 0.], [10., -10., 0.]]],
+                                       [[[0., 0., 10.], [10., 0., 10.]], [[0., -10., 10.], [10., -10., 10.]]]]]]])
+
+    corner_points_left = np.array([[[[[[[0., 0., 0.], [10., 0., 0.]], [[0., 10., 0.], [10., 10., 0.]]],
+                                      [[[0., 0., 10.], [10., 0., 10.]], [[0., 10., 10.], [10., 10., 10.]]]]]]])
+
+    # --------- Act----------
+    handedness_right = determine_corp_ijk_handedness(corner_points = corner_points_right)
+    handedness_left = determine_corp_ijk_handedness(corner_points = corner_points_left)
+
+    # --------- Assert----------
+    assert handedness_right == 'right'
+    assert handedness_left == 'left'
+
+
+# TODO: Find out if using a grid is too simple
+def test_determine_corp_extent(tmp_path):
+
+    # --------- Arrange----------
+    epc = os.path.join(tmp_path, 'grid.epc')
+    model = rq.new_model(epc)
+
+    # create a basic block grid
+    dxyz = (55.0, 65.0, 27.0)
+    grid = grr.RegularGrid(model, extent_kji = (4, 3, 2), title = 'concrete', origin = (0.0, 0.0, 1000.0), dxyz = dxyz)
+    grid.create_xml(add_cell_length_properties = True)
+    corner_points = grid.corner_points()
+    corner_points_reshaped = corner_points.reshape(1, 1, 24, 2, 2, 2, 3)
+    # --------- Act----------
+    [nk, nj, ni] = determine_corp_extent(corner_points = corner_points_reshaped)
+
+    # --------- Assert----------
+    assert [nk, nj, ni] == [4, 3, 2]
+
+
+@pytest.mark.parametrize(
+    'x_shift, y_shift, expected_result',
+    [(2, 0,
+      np.array([[[[[[[2.001, 0.001, 0.001], [12.001, 0.001, 0.001]], [[2.001, -10.001, 0.001], [12.001, -10.001, 0.001]]
```

Comment:
You might like to use the yapf disable comment to allow the expected result arrays to be laid out in the same way as the corner_points array.